ghost.py

```diff
@@ -986,6 +995,130 @@ def _get_document(self, key_name):
+                 region_name=None or os.environ.get('AWS_DEFAULT_REGION'),
+                 bucket_configuration=None):
+        """Initializes the storage client
+        :param db_path: The bucket name
+
+        Keyword arguments:
+        bucket_configuration
+        """
+        if not S3_EXISTS:
+            raise ImportError('boto3 and botocore must be installed first')
+        if not db_path:
+            raise GhostError('DB path (a bucket name) must be provided')
+        session = boto3.Session(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token,
+            profile_name=profile_name,
+            region_name=region_name)
+        self.client = session.client('s3')
+        self.db_path = db_path
+        if bucket_configuration:
+            self.bucket_configuration = bucket_configuration
+        else:
+            self.bucket_configuration = {'LocationConstraint': 'us-west-1'}
+
+    def init(self):
+        """Create a bucket.
+        """
+        self.client.create_bucket(
+            Bucket=self.db_path,
+            CreateBucketConfiguration=self.bucket_configuration)
+
+    def put(self, key):
+        """Insert the key
+        :return: Key name
+        """
+        self.client.put_object(
+            Body=json.dumps(key),
+            Bucket=self.db_path,
+            Key=key['name'])
+        return key['name']
+
+    def list(self):
+        """Lists the keys
+        :return: Returns a list of all keys (not just key names, but rather
+        the keys themselves).
+        """
+        response = self.client.list_objects_v2(Bucket=self.db_path)
+        if u'Contents' in response:
+            # Filter out everything but the key names
+            keys = [key[u'Key'] for key in response[u'Contents']]
+            keys_list = []
+
+            for key_name in keys:
+                key = self.get(key_name)
+                keys_list.append(key)
+
+            return keys_list
+        return []
+
+    def get(self, key_name):
+        """Gets the key.
+        :return: The key itself in a dictionary
+        """
+        def convert_dict_content_to_string(data):
+            """Recursively converts dictionary keys to strings."""
+            if isinstance(data, basestring):
+                return str(data)
+            elif not isinstance(data, dict):
+                return data
+            return dict((str(k), convert_dict_content_to_string(v))
+                        for k, v in data.items())
+
+        try:
+            obj = self.client.get_object(
+                Bucket=self.db_path,
+                Key=key_name)['Body'].read()
+            key = json.loads(obj)
+
+            return convert_dict_content_to_string(key)
+        except botocore.exceptions.ClientError as e:
+            error_code = e.response['Error']['Code']
+            if error_code == 'NoSuchKey':
```

Comment:
Well, all the other AWS client errors will be raised. I think any re-attempts and such should be performed outside the class.